test/Microsoft.IdentityModel.Xml.Tests/KeyInfoTests.cs

```diff
@@ -229,8 +228,115 @@ public static TheoryData<KeyInfoComparisonTheoryData> KeyInfoDataComparisonData
+
+                theoryData.ExpectedException.ProcessNoException(context);
+            }
+            catch (Exception ex)
+            {
+                theoryData.ExpectedException.ProcessException(ex, context);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<KeyInfoTheoryData> KeyInfoMatchesTheoryData
+        {
+            get
+            {
+                X509Certificate2 x509CertificateWithSki = CertificateHelper.LoadX509Certificate(KeyingMaterial.SelfSignedWithSKIExtension_Public);
+                X509ExtensionCollection extensions = x509CertificateWithSki.Extensions ?? throw new NotSupportedException("Extensions are null");
+                X509SubjectKeyIdentifierExtension skiExtension = extensions["2.5.29.14"] as X509SubjectKeyIdentifierExtension;
+                if (skiExtension == null)
+                    throw new NotSupportedException("X509SubjectKeyIdentifierExtension is null");
 
-#pragma warning restore CS3016 // Arrays as attribute arguments is not CLS-compliant
+                string ski = Convert.ToBase64String(skiExtension.RawData, 2, skiExtension.RawData.Length - 2);
+
+                TheoryData<KeyInfoTheoryData> theoryData = new TheoryData<KeyInfoTheoryData>();
+
+                theoryData.Add(new KeyInfoTheoryData("X509CertificateWithSKIExtension_UsingExtensibility")
+                {
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    KeyInfo = new CustomKeyInfo(ski),
+                    SecurityKey = new X509SecurityKey(x509CertificateWithSki),
+                    MatchesKey = true,
+                });
+
+                theoryData.Add(new KeyInfoTheoryData("X509CertificateWithoutSKIExtension_UsingExtensibility_NoMatch")
+                {
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    KeyInfo = new CustomKeyInfo(ski),
+                    SecurityKey = new X509SecurityKey(KeyingMaterial.X509Certificate2),
+                    MatchesKey = false,
+                });
+
+                theoryData.Add(new KeyInfoTheoryData("X509CertificateWithoutSKIExtension_UsingExtensibility")
+                {
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    KeyInfo = new CustomKeyInfo(KeyingMaterial.X509Certificate2),
+                    SecurityKey = new X509SecurityKey(KeyingMaterial.X509Certificate2),
+                    MatchesKey = true,
+                });
+
+                theoryData.Add(new KeyInfoTheoryData("X509CertificateWithSKIExtension")
+                {
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    KeyInfo = new KeyInfo(x509CertificateWithSki),
```

Comment:
Do we have a case for not using extensibility, Key info with SKI security key without SKI?